.github/workflows/ccpp.yml

```diff
@@ -0,0 +1,109 @@
+---
+name: Build CI
+on: [push]  # yamllint disable-line rule:truthy
+jobs:
+  build:
+    name: ${{ matrix.config.name }}
+    runs-on: ${{ matrix.config.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - {
+            name: "Windows MSVC",
+            enabled: 1,
+            os: windows-latest,
+            deps: "",
+            config: "cmake
+                    -B build
+                    -DCMAKE_TOOLCHAIN_FILE=C:/vcpkg/scripts/buildsystems/vcpkg.cmake
+                    -DWERROR=1
+                    -DCMAKE_BUILD_TYPE=Debug
+                    ",
+            build: "cmake --build build --config Debug",
+            # all tests work except "performance" (liblo bug?)
+            # this weird regex lets all those tests run, except performance
+            test: "ctest --output-on-failure --test-dir build -R '(i|s|t|g)'"
+          }
+          - {
+            name: "Ubuntu gcc",
+            enabled: 1,
+            os: ubuntu-latest,
+            deps: "sudo apt-get install liblo-dev",
+            config: "cd build && cmake -DWERROR=1 ..",
+            build: "cd build && make",
+            test: "cd build && ctest --output-on-failure"
+          }
+          - {
+            name: "Ubuntu clang+lld",
+            enabled: 1,
+            os: ubuntu-latest,
+            deps: "sudo apt-get install liblo-dev",
+            config: "cd build &&
+                    cmake
+                    -DWERROR=1
+                    -DCMAKE_C_COMPILER=clang
+                    -DCMAKE_CXX_COMPILER=clang++
+                    -DCMAKE_SHARED_LINKER_FLAGS='-fuse-ld=lld'
+                    -DCMAKE_EXE_LINKER_FLAGS='-fuse-ld=lld'
+                    ..",
+            build: "cd build && make",
+            test: "cd build && ctest --output-on-failure"
+          }
+          - {
+            name: "Ubuntu mingw32",
+            enabled: 1,
+            os: ubuntu-latest,
+            deps: "sudo apt-get install liblo-dev mingw-w64",
+            config: "cd build &&
+                    cmake
+                    -DWERROR=1
+                    CC=i686-w64-mingw32-gcc-win32
+                    CXX=i686-w64-mingw32-g++-win32
+                    ..",
+            build: "cd build && make",
+            test: "cd build && ctest --output-on-failure"
```

Comment:
It's a cross build, so are any tests getting run here?

Reply:
Good point. This build uses `CC=...` and `CXX=...`, but the `cmake` step says "Ignoring extra path from command line". I guess this is missing `-D...`, I will have to check. In any case, currently, this is a normal Linux build and test, I will try to fix that.

Reply:
Sounds good to me. No need to make that a blocker when the other improvements are all set.